airbyte-integrations/connectors/source-linkedin-ads/source_linkedin_ads/utils.py

```diff
@@ -205,42 +205,56 @@ def transform_targeting_criteria(
             }
 
     """
+
+    def unnest_dict(nested_dict: Dict) -> Iterable[List]:
+        """
+        Unnest the nested dict to simplify the normalization
+
+        EXAMPLE OUTPUT:
+            [
+                {"type": "some_key", "values": "some_values"},
+                ...,
+                {"type": "some_other_key", "values": "some_othre_values"}
+            ]
+        """
+
+        result = []
+        for key, value in nested_dict.items():
+            values = []
+            if isinstance(value, List):
+                if isinstance(value[0], str):
+                    values = value
+                elif isinstance(value[0], Dict):
+                    for v in value:
+                        values.append(v)
+            elif isinstance(value, Dict):
+                values.append(value)
+            result.append({"type": key, "values": values})
+        yield from result
+
+    # get the target dict from record
     targeting_criteria = record.get(dict_key)
+
     # transform `include`
     if "include" in targeting_criteria:
         and_list = targeting_criteria.get("include").get("and")
-        for id, and_criteria in enumerate(and_list):
-            or_dict = and_criteria.get("or")
-            for key, value in or_dict.items():
-                values = []
-                if isinstance(value, list):
-                    if isinstance(value[0], str):
-                        values = value
-                    elif isinstance(value[0], dict):
-                        for v in value:
-                            values.append(v)
-                elif isinstance(key, dict):
-                    values.append(key)
-                # Replace the 'or' with {type:value}
-                record["targetingCriteria"]["include"]["and"][id]["type"] = key
-                record["targetingCriteria"]["include"]["and"][id]["values"] = values
-                record["targetingCriteria"]["include"]["and"][id].pop("or")
+        updated_include = {"and": []}
+        for k in and_list:
+            or_dict = k.get("or")
+            unnested = unnest_dict(or_dict)
+            for j in unnested:
+                updated_include["and"].append(j)
```

Comment:
please rename `unnested`  to similar `unnested_iter` for clarity. Or call the function `unnest_dict` into the next `for` operator directly.

Reply:
We will call lit directly, within the `for`